### PR TITLE
Snow 1639660 cleanup "todo in 3.0" comments

### PR DESCRIPTION
**Pull request by sfc-gh-pczajka**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [ ] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Cleanup "TODO in 3.0" comments:
 * remove Anaconda channel fallback when fetching info about available anaconda packages
 * remove `ENABLE_PROJECT_DEFINITION_V2` feature flag

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -27,6 +27,9 @@
   * `snow snowpark deploy` uploads all artifacts created during build step. Dependencies zip is upload once to
      every Snowpark stage specified in project definition.
   * The changes are compatible with V1 projects definition though the result state (file layout) is different.
+  * `snow snowpark package` commands no longer fallback to Anaconda Channel metadata when fetching available packages
+    info fails.
```
Comment (sfc-gh-astus) [inline RELEASE-NOTES.md:31]: Is this related to previous line?

Comment (sfc-gh-pczajka) [reply RELEASE-NOTES.md:31]: Yes, "fetching available packages info fails"

**Comment by sfc-gh-cgorrie** [comment]
@sfc-gh-pczajka there are still test references to `enable_project_definition_v2`

File: RELEASE-NOTES.md (hunk 1 above)
Comment (sfc-gh-astus) [reply RELEASE-NOTES.md:31]: Ok. Could you move to one line?

Comment (sfc-gh-pczajka) [reply RELEASE-NOTES.md]: Done

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -27,6 +27,8 @@
   * `snow snowpark deploy` uploads all artifacts created during build step. Dependencies zip is upload once to
      every Snowpark stage specified in project definition.
   * The changes are compatible with V1 projects definition though the result state (file layout) is different.
+  * `snow snowpark package` commands no longer fallback to Anaconda Channel metadata when fetching available packages info fails.
+* `ENABLE_PROJECT_DEFINITION_V2` feature flag is removed. Project definition V2 is always enabled.
```
Comment (sfc-gh-turbaszek) [inline RELEASE-NOTES.md:31]: This is probably not necessary, I don't think we ever officially said that there's something like this.

Comment (sfc-gh-pczajka) [reply RELEASE-NOTES.md]: Removed

**Review by sfc-gh-astus — DISMISSED** [review]

**Review by sfc-gh-turbaszek — APPROVED** [review]

**Review by sfc-gh-turbaszek — DISMISSED** [review]

**Review by sfc-gh-astus — APPROVED** [review]

**Review by sfc-gh-jsikorski — DISMISSED** [review]

**Review by sfc-gh-turbaszek — DISMISSED** [review]

**Review by sfc-gh-astus — APPROVED** [review]

**Review by sfc-gh-melnacouzi — APPROVED** [review]
lgtm
